### PR TITLE
Adds ability to auto approve and merge Dependabot PRs

### PR DESCRIPTION
**Pull request by allez-allez-allez**
This PR will add the ability to auto-approve and auto-merge pull requests created by Depenedabot.
Utilizing GitHub action to auto-approve and using Mergify.io to auto-merge the pull requests.

Configuration steps for GitHub Action:
* Signup for GitHub Actions Beta

Configuration steps for Mergify.io:
* Grant mergify access to the repo. This step is completed, but if it needs to be re-configured, you can access this [site](https://mergify.io/dashboard) to do so.

### CODE REVIEW

File: .github/workflows/approve-dependabot-pr.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+jobs:
+  auto-approve-dependabot-pr:
+    runs-on: ubuntu-latest
+    steps:
+      - name: auto approve dependabot pull requests
+        uses: hmarr/auto-approve-action@master
+        if: github.actor == 'dependabot[bot]' || github.actor == 'dependabot-preview[bot]'
+        with:
+          github-token: "${{ secrets.DEPENDABOT_TOKEN }}"
```
Comment (lorengordon) [inline .github/workflows/approve-dependabot-pr.yml]: Can we just use `secrets.GITHUB_TOKEN`? That does not require a custom token to be created... 

https://help.github.com/en/articles/virtual-environments-for-github-actions#github_token-secret

Comment (allez-allez-allez) [reply .github/workflows/approve-dependabot-pr.yml]: I tried using `GITHUB_TOKEN`, but it wouldn't trigger the mergify to auto-merge.

When using `GITHUB_TOKEN`. the pr is approved by `github-action[bot]`, somehow mergify doesn't recognize it as an approved pr so it failed the condition check.

I will test it again, just to confirm.

Comment (lorengordon) [reply .github/workflows/approve-dependabot-pr.yml]: Let's start with opening an issue with mergify and see if they can provide some guidance...

Comment (lorengordon) [reply .github/workflows/approve-dependabot-pr.yml]: Ahh, can you add `github-action[bot]` as a collaborator with write privs to that repo?

Comment (allez-allez-allez) [reply .github/workflows/approve-dependabot-pr.yml]: There were no issues at all. submitted an [issue](https://github.com/Mergifyio/mergify-engine/issues/514) to mergify and got a response instantly.

`
If you are using branch protection, you can safely remove - "#approved-reviews-by>=1". When merge action is run, Mergify always wait the branch protection to be validated before merging the PR.
`
Tested with `GITHUB_TOKEN` and it works. I have updated this pr to reflect the suggestions.

File: .mergify.yml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+pull_request_rules:
+  - name: automatic merge for approved Dependabot pull requests
+    conditions:
+      - author=dependabot-preview[bot]
```
Comment (lorengordon) [inline .mergify.yml]: we have two conditions for the actor/author on the github action, but only one here... can we check for both here?

```
if: github.actor == 'dependabot[bot]' || github.actor == 'dependabot-preview[bot]'
```

Comment (allez-allez-allez) [reply .mergify.yml]: we can use a regular expression to match the author name like this.
```yaml
    conditions:
      - author~=\b(\w*dependabot\w*)\b
```
not sure if this is the best regular epression though.

Comment (lorengordon) [reply .mergify.yml]: Does the regex processor support `|` as "or"? e.g. something like `author~=dependabot\[bot\]|dependabot-preview\[bot\]`

Comment (allez-allez-allez) [reply .mergify.yml]: The way you have it works. Think it's ready.

**Pull request merged**